product/synquacer/include/qspi_api.h

```diff
@@ -0,0 +1,249 @@
+/*
+ * Arm SCP/MCP Software
+ * Copyright (c) 2021-2022, Arm Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef QSPI_API_H
+#define QSPI_API_H
+
+#include <fwk_id.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+/*
+ * \brief qspi command structure
+ */
+struct qspi_command {
+    uint8_t code;
+    struct len {
+        uint8_t code_byte;
+        uint8_t addr_byte;
+        uint8_t alt_byte;
+        uint8_t dummy_cycle;
+        uint8_t data_byte;
+    } len;
+    struct alt {
+        uint8_t data[4];
+        bool is_nibble;
+    } alt;
+    struct io {
+        uint8_t code_bit;
+        uint8_t addr_bit;
+        uint8_t data_bit;
+    } io;
+};
+
+/*
+ * \brief qspi command generator macro definitions
+ */
+#define GEN_COMMAND( \
+    CODE, \
+    CODE_BYTE, \
+    ADDR_BYTE, \
+    ALT_BYTE, \
+    DUMMY, \
+    DATA_BYTE, \
+    ALT_0, \
+    ALT_1, \
+    ALT_2, \
+    ALT_3, \
+    NIBBLE, \
+    IO_CODE, \
+    IO_ADDR, \
+    IO_DATA) \
+    { \
+        .code = CODE,                 \
+        .len = {                      \
+            .code_byte = CODE_BYTE,   \
+            .addr_byte = ADDR_BYTE,   \
+            .alt_byte = ALT_BYTE,     \
+            .dummy_cycle = DUMMY,     \
+            .data_byte = DATA_BYTE,   \
+        },                            \
+        .alt = {                      \
+            .data = {                 \
+                ALT_0,                \
+                ALT_1,                \
+                ALT_2,                \
+                ALT_3                 \
+            },                        \
+            .is_nibble = NIBBLE,      \
+        },                            \
+        .io = {                       \
+            .code_bit = IO_CODE,      \
+            .addr_bit = IO_ADDR,      \
+            .data_bit = IO_DATA,      \
+        }, \
+    }
+#define COMMAND_EMPTY \
+    ((struct qspi_command) \
+        GEN_COMMAND(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, false, 0, 0, 0))
+#define COMMAND_TYPE_CODE(CODE) \
+    ((struct qspi_command) \
+        GEN_COMMAND(CODE, 1, 0, 0, 0, 0, 0, 0, 0, 0, false, 1, 0, 1))
+#define COMMAND_TYPE_CODE_DATA(CODE) \
+    ((struct qspi_command) \
+        GEN_COMMAND(CODE, 1, 0, 0, 0, 1, 0, 0, 0, 0, false, 1, 0, 1))
+
+#define COMMAND_TYPE_READ(CODE, ADDR, DUMMY, IO_CODE, IO_ADDR, IO_DATA) \
+    ((struct qspi_command) GEN_COMMAND( \
+        CODE, \
+        1, \
+        ADDR, \
+        0, \
+        DUMMY, \
+        1, \
+        0, \
+        0, \
+        0, \
+        0, \
+        false, \
+        IO_CODE, \
+        IO_ADDR, \
+        IO_DATA))
+#define COMMAND_TYPE_WRITE(CODE, ADDR, IO_CODE, IO_ADDR, IO_DATA) \
```

Comment:
I suggest using QSPI_COMMAND_TYPE_WRITE or QSPI_CMD_TYPE_WRITE so it's more specific to qspi

Reply:
I will add QSPI_ prefix.